spec/rubocop/cop/lint/shadowing_outer_local_variable_spec.rb

```diff
@@ -322,6 +338,16 @@ def some_method
     end
   end
 
+  context 'when the same variable name as a block variable is used in return value assignment of method' do
+    it 'does not register an offense' do
+      expect_no_offenses(<<~RUBY)
+        def some_method
+          foo = bar { |foo| baz(foo) }
+        end
+      RUBY
+    end
+  end
```

Comment:
This is similar to the test right above, only without the condition.

Side note: personally, I think this should be an offense since it's confusing. Should I open another PR for that?

Reply:
> Side note ...

I guess this is true but I also feel there is no ambiguity in this case. By the time the block variable is defined, the assignment variable will only ever be `nil` (unless defined elsewhere before). Locally it will also be very close to the block itself.

Reply:
I agree. No need to turn this to an offense.